geth/api/backend.go

```diff
@@ -77,120 +80,146 @@ func (m *StatusBackend) IsNodeRunning() bool {
 }
 
 // StartNode start Status node, fails if node is already started
-func (m *StatusBackend) StartNode(config *params.NodeConfig) (<-chan struct{}, error) {
-	m.Lock()
-	defer m.Unlock()
-
-	if m.nodeReady != nil {
-		return nil, node.ErrNodeExists
-	}
+func (m *StatusBackend) StartNode(config *params.NodeConfig) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.startNode(config)
+}
 
-	nodeStarted, err := m.nodeManager.StartNode(config)
+func (m *StatusBackend) startNode(config *params.NodeConfig) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("node crashed on start: %v", err)
+		}
+	}()
+	err = m.nodeManager.StartNode(config)
 	if err != nil {
-		return nil, err
+		switch err.(type) {
+		case node.RPCClientError:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: node.ErrRPCClient.Error(),
+				},
+			})
+		case node.EthNodeError:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: fmt.Errorf("%v: %v", node.ErrNodeStartFailure, err).Error(),
+				},
+			})
+		default:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: err.Error(),
+				},
+			})
+		}
+		return err
 	}
-
-	m.nodeReady = make(chan struct{}, 1)
-	go m.onNodeStart(nodeStarted, m.nodeReady) // waits on nodeStarted, writes to backendReady
-
-	return m.nodeReady, err
+	signal.Send(signal.Envelope{
+		Type:  signal.EventNodeStarted,
+		Event: struct{}{},
+	})
+	m.onNodeStart()
```

Comment:
We should move it to `StartNode` method as it will be easier to understand the flow will shorter call stack.

Reply:
it became rather short after i removed duplicated signals, will leave it here for now